### ISSUE TITLE
Only "not visible by other" contacts displayed

### ISSUE THREAD

**Issue by wicako** [SimpleMobileTools/Simple-Contacts#370]
Updating to version 6.3.0, only the "not visible by other apps" contacts are displayed. 
The option to display "public" contacts (under filters) is gone.
Same problem with 6.2.0.
Downgrading two steps back to version 6.1.2 bring back the "public" contacts.

Using a moto e5 under android 8. Let me know if I can help diagnose.
Thanks!

**Comment by michaelof** [comment]
@wicako
You might want to take a look at my issue #351. Sound the same, if your "public" contacts are meaning the "phone" contacts.

This issue is already closed and fixed, and I can confirm that all potential contact sources are shown as filters in my 6.3.0.

Interesting...

**Comment by wicako** [comment]
@michaelof 
Yes, by "public" I meant "phone" contacts.

Indeed, it sounds related to the issue you raised previously. I tried to replicate your test of creating a new contact from the stock application, but no joy. Still only "not visible by other apps" contacts available under filters.

Discovered something else. Using Simple Contacts (6.3.0) to create a new contact, the entry will not show up in Simple Contacts but it will in the stock app. That said, this comes from multiple 6.1.2 to 6.3.0 reinstall... so maybe there is some remaining settings causing that.

Hope this can help narrow things down. I can do further testing.

**Comment by tibbi** [comment]
have you tried reinstalling the ap?

**Comment by wicako** [comment]
Hi Tibor,
first and foremost, thank you very much for developing all your apps!

Just tried to completely uninstall Simple Contacts 6.1.2  -- the last version displaying the "phone" contacts in my case -- before installing 6.3.0. Same problem, no contacts displayed and no option to enable them in filters (only option available is phone storage "not visible by other apps")

I created a new entry using Simple Contacts and it only shows up in the stock contact app. 
Looks like I can write to the shared "phone" database, but cannot read it back from Simple Contacts.

Let me know if you want me to test something else.
Thanks!

**Comment by gouge34** [comment]
Have you solved the problem yet?

**Comment by I-Al-Istannen** [comment]
I have a similiar problem. I see all three sources (DEVICE, Signal, phone storage), but it displays no contact.

### Down the rabbit hole

After stepping through the source, the account name seems to be null:
![image](https://user-images.githubusercontent.com/20284688/56609701-0cb0cf00-660e-11e9-9f21-d5429f3d9d8b.png)
This is then corrected to `""` by the code, but therefore it *does not match any source filter* and is subsequently dropped.

I am not sure why that column is null, the cursor dump looks like this:
```
{
       contact_id=574
       raw_contact_id=388
       data4=
       data2=ADAC
       data5=
       data3=Copilot
       data6=
       photo_uri=null
       photo_thumb_uri=null
       starred=0
       account_name=null
       account_type=null
}
```

Maybe allowing an "all" option instead of just the known sources would be nice as a workaround? Or just not filtering at all if all sources are selected?

I sadly do not have enough knowledge of the contact system to judge *why* that thing is null, so there might be a better option.

I will just recompile it without the filtering for me, but just @ me if I can be of some help :)

Have a nice day!

**Comment by tibbi** [comment]
It used to be like that, but that caused some other issues. Apparently I dont understand the contacts system well either, it is just another of those things that can be done in so many different ways depending on the device.

**Comment by I-Al-Istannen** [comment]
Hm. What were the problems. I can only imagine that everybody does it differently, my phone is running LOS 15.1 if that helps.

If the problems were not too severe, changing the filter to *exclude* the unchecked ones instead of including the checked ones might work?

**Comment by tibbi** [comment]
that was actually the initial implementation too. I dont remember the specific issues now, will take a look at it again someday.

**Comment by I-Al-Istannen** [comment]
Oh well. My current version is working for me now (just show everything if all are checked), so I am set. Still a bit of a weird issue, but it does indeed not seem nice to fix.

I wish you the best if you have a closer look at it some day, thank you ;)

For now just enjoy the present one!

**Comment by lightdot** [comment]
I just had a problem getting contacts do display, search led me here (perhaps related to #351 too).

AOSP Extended (Android 9), OS Contacts app (com.android.contacts 1.7.31) disabled, with cache and data deleted (!).

Adding a contact in Simple Contacts resulted in a contact that was only visible within the original OS dialer (Phone app, com.android.dialer 19.0) but not in Simple Contacts.

As if the contact was added to the Android Contacts database, but the database or the contact was unknown to the Simple Dialer.

By selecting this contact in the OS dialer, I was able to open it in Simple Contatcs for both viewing and editing, but it still did not show up the Simple Contacts contact list.

When viewing the contact, the field where "DEVICE" should be displayed was empty, as if the contact was created without any specification if it is a standard contact in the Android contact's database or not.

Among the contacts filters the only choice was "Phone storage (not visible by other apps)".

At this point, I have re-enabled the OS Contacts app and created a contact in it, this has changed the Simple Contacts behaviour.

Now I have a choice between "DEVICE" and "Phone storage (not visible by other apps)" in Simple Contacts and if I create a "DEVICE" contact, it is visible in all mentioned applications.

I don't know enough about Android, but my hunch is that this is related to the Android Contacts database initialization and/or the inability to mark the contact as "DEVICE" contact in the first place.

Anyway, for whomever stumbles on the same issue, it seems that the contacts database needs to be initialized by the OS Contacts app for the Simple Contacts to start working as expected.

Hope this helps.
